src/XamlX/Transform/XamlContextBase.cs

```diff
@@ -12,6 +12,10 @@ class XamlContextBase
     {
         private readonly Dictionary<Type, object> _items = new Dictionary<Type, object>();
```

Comment:
Maybe I am wrong but isn't the count of items here always equal to your _I'dCount?